chanacceptor/zeroconfacceptor.go

```diff
@@ -0,0 +1,53 @@
+package chanacceptor
+
+// ZeroConfAcceptor wraps a regular ChainedAcceptor. If no acceptors are in the
+// ChainedAcceptor, then Accept will reject all channel open requests. This
+// should only be enabled when the zero-conf feature bit is set and is used to
+// protect users from a malicious counter-party double-spending the zero-conf
+// funding tx.
+type ZeroConfAcceptor struct {
```

Comment:
why cant `ZeroConfAcceptor` just be another implementation of `ChannelAcceptor` that we would add to the existing ChainedAcceptor?

Reply:
`ZeroConfAcceptor` logic is:
- if no acceptors exist, fail
- otherwise, use those acceptors
so I think it needs to wrap the `ChainedAcceptor` and have awareness of all existing acceptors rather than being another one of the acceptors in `ChainedAcceptor`